.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,67 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '44 12 * * 4'
```

Comment:
Why this specific schedule? Why not just trigger on pull requests and pushes?

Reply:
It does run part of every PR here is an example  https://github.com/ossf/scorecard/actions?query=workflow%3ACodeQL and more information https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onschedule

Reply:
Sorry I wasn't clear, I saw that it runs on PRs but wondering why it additionally triggers on a schedule and why this particular schedule.

Reply:
TBH, that is the standard codeql template from GitHub. I have been using that template.

Reply:
As I was thinking about this. It also makes sense to have a cron for this because the upstream(codeql) has new updates and those new checks have to be run on a scheduled timeline.

Reply:
Good point :+1: